### PR TITLE
[Doc] Logging: Add Fluent Bit sidecar and Grafana Loki to Persist Ray logs

### PR DESCRIPTION
**Pull request by win5923**
## Why are these changes needed?
- Add a tutorial on how to set up Fluent Bit sidecar logging with Grafana Loki on Kubernetes. The FluentBit sidecar container redirects logs from `/tmp/ray/session_latest/logs` to Grafana Loki.
- Using [downward API](https://kubernetes.io/docs/concepts/workloads/pods/downward-api/) to fetch pod metadata labels. This allows users to configure Fluent Bit to assign logs based on specific labels, such as `ray.io/cluster`
```yaml
env:
  - name: POD_LABELS
    valueFrom:
      fieldRef:
        fieldPath: metadata.labels['ray.io/cluster']
```
- Users can modify the Fluent Bit ConfigMap's `Labels` to associate logs with different RayCluster / RayJob CRs.
```
Labels RayCluster=raycluster-xxxx
Labels RayJob=rayjob-xxxx
```

## Related issue number
https://github.com/ray-project/kuberay/issues/2515

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/kubernetes/configs/ray-cluster.log.yaml
Comment (kevin85421) [inline doc/source/cluster/kubernetes/configs/ray-cluster.log.yaml:19]: Can we use K8s labels (https://docs.fluentbit.io/manual/pipeline/filters/kubernetes) instead to dynamically set up the log label?

File: doc/source/cluster/kubernetes/configs/ray-cluster.log.yaml
Comment (kevin85421) [inline doc/source/cluster/kubernetes/configs/ray-cluster.log.yaml:12]: Can you open a PR to add a RayCluster YAML to the KubeRay repo and reference the YAML in this doc?

Example:
https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/configuring-autoscaling.html#step-3-create-a-raycluster-custom-resource-with-autoscaling-enabled

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md
Comment (kevin85421) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:65]:
```suggestion
* Assign logs to a specific tenant using the `Tenant_ID` field, enabling multi-tenancy in Grafana Loki. In this example, it matches the `test` tenant specified in the Grafana Loki datasource configuration.
```

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md
Comment (kevin85421) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:132]: delete the YAML in Ray repo and add a YAML in KubeRay repo instead

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md
Comment (kevin85421) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:150]:
```suggestion
Finally, use a LogQL query to view logs for a specific RayCluster or RayJob, and filter by the `RayCluster_name`:
```

**Review by kevin85421 — APPROVED** [review]
I will polish the doc after this PR is merged

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:59]:
```suggestion
* Send logs to a Grafana Loki endpoint.
```

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:60]:
```suggestion
* Add metadata to the logs for filtering by labels, for example, `RayCluster`.
```

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:74]:
```suggestion
- The [Kubernetes downward API][KubernetesDownwardAPI] populates the `POD_LABELS` variable used in the `FILTER` section. It pulls the label from the pod's metadata label `ray.io/cluster`, which is defined in the Fluent Bit sidecar container's environment.
```

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:75]:
```suggestion
- The `tenant_id` field allows you to assign logs to different tenants. In this example, Fluent Bit sidecar sends the logs to the `test` tenant. You can adjust this configuration to match the tenant ID set up in your Grafana Loki instance, enabling multi-tenancy support in Grafana.
```

**Review by angelinalg — COMMENTED** [review]
Feel free to correct any inaccuracies in my rewrites. The suggestions are mostly removing passive voice.

**Review by angelinalg — APPROVED** [review]
Approving so as to not block.

**Comment by win5923** [comment]
Thanks for @kevin85421 and @angelinalg's reviews. I've completed the style fixes.

**Pull request merged**
